src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,142 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private int size;
+    private final int defaultInitialCapacity = 16;
+    private final float defaultLoadFactor = 0.75f;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        threshold = 0;
+        size = 0;
+    }
+
+    private int hash(K key) {
+        return key == null ? 0 : key.hashCode();
+    }
+
+    private Node<K, V>[] resize() {
+        final Node<K, V>[] oldTab = table;
+        int oldCap = ((table == null) ? 0 : table.length);
+        int oldThr = threshold;
+        int newThr;
+        int newCap;
+        if (oldCap > 0) {
+            newCap = oldCap << 1;
+            newThr = oldThr << 1;
+        } else {
+            newThr = (int) (defaultInitialCapacity * defaultLoadFactor);
+            newCap = defaultInitialCapacity;
+        }
+        threshold = newThr;
+        Node<K,V>[] newTab = (Node<K,V>[])new Node[newCap];
+        table = newTab;
+        if (oldTab != null) {
+            for (int i = 0; i < oldTab.length; i++) {
+                Node<K, V> node;
+                if ((node = oldTab[i]) != null) {
+                    oldTab[i] = null;
+                    if (node.next == null) {
+                        newTab[node.hash & (newCap - 1)] = node;
+                    } else {
+                        while (node != null) {
+                            final Node<K, V> current = node;
+                            node = node.next;
+                            putNode(current, newTab);
+                        }
+                    }
+                }
+            }
+        }
+        return newTab;
+    }
+
+    private void putNode(Node<K, V> current, Node<K, V>[] newTab) {
+        current.next = null;
+        int index;
+        if ((newTab[index = (current.hash & (newTab.length - 1))]) == null) {
+            newTab[index] = current;
+        } else {
+            Node<K, V> tn = newTab[index];
+            K key = current.key;
+            if (tn.hash == current.hash && (key == tn.key || (key != null && key.equals(tn.key)))) {
+                newTab[index].value = current.value;
+            } else {
+                while (tn.next != null) {
+                    tn = tn.next;
+                }
+                tn.next = current;
+            }
+        }
+    }
+
+    public void putVal(int hash, K key, V value) {
+        int n;
+        int index;
+        if (table == null || (n = table.length) == 0) {
```

Comment:
imho: it's better to init table in the constructor like = new Node[INITIALI_CAPACIT], this way we won't need this check here. The check improves performance if you create a lot of hash map objects and don't use them but for now, it's overengineering